src/context.js

```diff
@@ -0,0 +1,66 @@
+// @flow
+import * as React from 'react';
+import {createPlugin} from 'fusion-core';
+import type FusionApp, {FusionPlugin, Middleware} from 'fusion-core';
+import type {Element} from 'react';
+
+export const FusionContext = React.createContext<any>({});
+export const ServiceContext = React.createContext<any>(() => {});
+
+type ReturnsType<T> = () => T;
```

Comment:
Needed a mechanism to support `Token.optional` since flow considers it different from `Token`. This technically allows someone to pass in something that isn't a token. The usage of this function is so straightforward that I don't think this is a huge issue.